pkg/kv/kvserver/client_raft_helpers_test.go

```diff
@@ -172,6 +172,50 @@ func (h *unreliableRaftHandler) HandleDelegatedSnapshot(
 	return h.IncomingRaftMessageHandler.HandleDelegatedSnapshot(ctx, req)
 }
 
+type filterRaftHandlerFuncs struct {
+	// If set to nil, all incoming messages are dropped. If non-nil, returning
+	// true can prevent the message from being dropped.
+	filterReq func(*kvserverpb.RaftMessageRequest) bool
+	// If set to nil, no additional processing is applied to outgoing messages
+	// from the sender's end. If non-nil, returning true can allow the additional
+	// processing.
+	filterReqSent func(*kvserverpb.RaftMessageRequest) bool
+}
+
+// filterRaftHandler applies the filter functions within filterRaftHandlerFuncs
+// to the incoming and outgoing messages. It ensures that only messages with
+// filters that evaluate to true are forwarded to the underlying store
+// interface.
+type filterRaftHandler struct {
```

Comment:
Is this accidentally reinventing the wheel? Check `unreliableRaftHandler`, which has various interceptors. If that one doesn't fit your need, please extend it instead of adding a new thing.

Reply:
Discussed this over slack - The handler unreliableRaftHandler has a specific field range ID that is designated to drop messages towards this RangeID. And the filter functions they have only avoids dropping messages to this specific range ID. It doesn't drop or avoid dropping anything to other rangeIDs. I was considering refactoring `unreliableRaftHandler` to use the filterHandler I’m adding but thought it would be better to be done in a separate PR.

Anyway, I don't think this would be a problem since we’ve decided to just get rid of large test cases altogether for metrics update and adopt unit tests.